src/sortStudents.ts

```diff
@@ -1,16 +1,54 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+const getAverageGrade = (grades: number[]): number => (
+  grades.reduce((sum, grade) => sum + grade, 0) / grades.length);
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyOFStudents: Student[] = [...students];
+
+  copyOFStudents.sort((studentA: Student, studentB: Student) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? studentA[sortBy].localeCompare(studentB[sortBy])
+          : studentB[sortBy].localeCompare(studentA[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(studentA[sortBy]) - Number(studentB[sortBy])
+          : Number(studentB[sortBy]) - Number(studentA[sortBy]);
 
+      default:
```

Comment:
```suggestion
      case SortType.AverageGrade:
```
Not sure it's a good idea to use default instead of specific case. At least in terms of code clarity. According to documentation, default is optional, so you can just ommit it.

Reply:
Yeap, you are right. Let's use explicit case `case SortType.AverageGrade:`
By default, we don't sort or just throw an error